### PR TITLE
Add document for all the commands

### PR DESCRIPTION
**Pull request by pradeepitm12**
This add document for the commands in cli.

Signed-off-by: Pradeep Kumar <pradkuma@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```
release-note
```

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# tkn
+`tkn` is the Tekton pipeline command line interface (CLI)
+
+## Getting Started
+
+### Installing `tkn`
+
+You can grab the latest nightly binary executalbe for :
```
Comment (vdemeester) [inline docs/README.md]: We will use releases :angel:

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# tkn
```
Comment (siamaksade) [inline docs/README.md]: This content should probably be in the README.md in the root of the repo

File: docs/README.md (hunk 1 above)
Comment (abrennan89) [inline docs/README.md]: typo on 'executable'

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+### Installing `tkn`
+
+You can grab the latest nightly binary executalbe for :
+# Note:- Need to update below links.
+* [Max OS X](The/URL/to/binary)
+* [Linux AMD 64](The/URL/to/binary)
+* [Windows AMD 64](The/URL/to/binary)
+
+## Need to add installation steps.
+
```
Comment (siamaksade) [inline docs/README.md]: Should there be a section with the list of commands, linked to their respective doc page?

**Comment by vdemeester** [comment]
```
I0607 12:53:41.856] ----------------------------------------------
I0607 12:53:41.857] ---- Checking links in the markdown files ----
I0607 12:53:41.857] ----------------------------------------------
I0607 12:53:42.170] docs/README.md
I0607 12:53:42.170] 	ERROR	The/URL/to/binary
I0607 12:53:42.170] 		Stat docs/The/URL/to/binary: no such file or directory
```
@pradeepitm12  ahah, let's add a link to the `README.md` or something for now :)

File: docs/README.md (hunk 3 above)
Comment (pradeepitm12) [reply docs/README.md]: Adding all the commands will make it large which will definitely grow soon.
How about adding links to all the parent commands docs like pipeline, pipelinerun, task, taskrun which further have links to their child commands?

**Comment by siamaksade** [comment]
Sounds reasonable :)

File: docs/cmd/tkn_completion.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
+## tkn completion
+
+Output shell completion code (default Bash)
```
Comment (Preeticp) [inline docs/cmd/tkn_completion.md]: Outputs

File: docs/cmd/tkn_pipeline.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
+## tkn pipeline
+
+Pipeline command group
+
```
Comment (Preeticp) [inline docs/cmd/tkn_pipeline.md]: How about: Parent command of the Pipeline command group .

File: docs/cmd/tkn_pipeline_describe.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
+## tkn pipeline describe
+
+Describe pipeline.
```
Comment (Preeticp) [inline docs/cmd/tkn_pipeline_describe.md]: Describes the

File: docs/cmd/tkn_pipeline_list.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
+## tkn pipeline list
+
+List pipelines in a namespace.
```
Comment (Preeticp) [inline docs/cmd/tkn_pipeline_list.md]: Lists the...

File: docs/cmd/tkn_pipelinerun.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
+## tkn pipelinerun
+
+Pipelinerun command group
+
```
Comment (Preeticp) [inline docs/cmd/tkn_pipelinerun.md]: Parent command of the Pipelinerun command group

File: docs/cmd/tkn_pipelinerun_describe.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
+## tkn pipelinerun describe
+
+Describe pipelinerun.
```
Comment (Preeticp) [inline docs/cmd/tkn_pipelinerun_describe.md]: Describes the...

File: docs/cmd/tkn_pipelinerun_list.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
+## tkn pipelinerun list
+
+List pipelineruns in a namespace.
```
Comment (Preeticp) [inline docs/cmd/tkn_pipelinerun_list.md]: Lists the

File: docs/cmd/tkn_pipelinerun_logs.md [hunk 11]
```diff
@@ -0,0 +1,31 @@
+## tkn pipelinerun logs
+
+Logs of pipelinerun in a namespace.
```
Comment (Preeticp) [inline docs/cmd/tkn_pipelinerun_logs.md]: Lists the logs for the...

File: docs/cmd/tkn_task.md [hunk 12]
```diff
@@ -0,0 +1,37 @@
+## tkn task
+
+Task command group
+
```
Comment (Preeticp) [inline docs/cmd/tkn_task.md]: Parent command of the Task command group.

File: docs/cmd/tkn_task_list.md [hunk 13]
```diff
@@ -0,0 +1,38 @@
+## tkn task list
+
+List tasks in a namespace.
```
Comment (Preeticp) [inline docs/cmd/tkn_task_list.md]: Lists the...

File: docs/cmd/tkn_taskrun.md [hunk 14]
```diff
@@ -0,0 +1,38 @@
+## tkn taskrun
+
+Taskrun command group
+
```
Comment (Preeticp) [inline docs/cmd/tkn_taskrun.md]: Parent command of the Taskrun command group

File: docs/cmd/tkn_taskrun_list.md [hunk 15]
```diff
@@ -0,0 +1,38 @@
+## tkn taskrun list
+
+List taskruns in a namespace.
```
Comment (Preeticp) [inline docs/cmd/tkn_taskrun_list.md]: Lists the

File: docs/cmd/tkn_taskrun_logs.md [hunk 16]
```diff
@@ -0,0 +1,32 @@
+## tkn taskrun logs
+
+Logs of taskrun in a namespace.
```
Comment (Preeticp) [inline docs/cmd/tkn_taskrun_logs.md]: Lists the logs for the taskrun...

File: docs/cmd/tkn_version.md [hunk 17]
```diff
@@ -0,0 +1,17 @@
+## tkn version
+
+Prints tekton cli version
```
Comment (Preeticp) [inline docs/cmd/tkn_version.md]: Prints the

File: docs/cmd/tkn_pipeline_describe.md [hunk 18]
```diff
@@ -0,0 +1,38 @@
...
+
+### Synopsis
+
+Describe pipeline.
```
Comment (joaedwar) [inline docs/cmd/tkn_pipeline_describe.md]:
```suggestion
Describes the pipeline.
```

File: docs/cmd/tkn_pipeline_describe.md [hunk 19]
```diff
@@ -0,0 +1,38 @@
...
+### Aliases
+
+```
+describe, desc
+```
+
+### Options
+
+```
+--allow-missing-template-keys   If true, ignore any errors in templates when a field or map key is missing in the template. Only applies to golang and jsonpath output formats. (default true)
```
Comment (joaedwar) [inline docs/cmd/tkn_pipeline_describe.md]:
```suggestion
-allow-missing-template-keys   If true, ignores any errors in a template when a field or map key is missing. Only applies to golang and jsonpath output formats. (default true)
```

File: docs/cmd/tkn_pipeline_list.md [hunk 20]
```diff
@@ -0,0 +1,38 @@
...
+### Aliases
+
+```
+list, ls
+```
+
+### Options
+
+```
+--allow-missing-template-keys   If true, ignore any errors in templates when a field or map key is missing in the template. Only applies to golang and jsonpath output formats. (default true)
```
Comment (joaedwar) [inline docs/cmd/tkn_pipeline_list.md]:
```suggestion
-allow-missing-template-keys   If true, ignores any errors in a template when a field or map key is missing. Only applies to golang and jsonpath output formats. (default true)
```

File: docs/cmd/tkn_pipelinerun.md [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+```
+pipelinerun, pr, pipelineruns
+```
+
+### Available Commands
+
+```
+describe    Describe a pipelinerun in a namespace
+list        Lists pipelineruns in a namespace
+logs        Show the logs of PipelineRun
```
Comment (joaedwar) [inline docs/cmd/tkn_pipelinerun.md]:
```suggestion
logs        Display PipelineRun logs
```

File: docs/cmd/tkn_pipelinerun.md [hunk 22]
```diff
@@ -0,0 +1,41 @@
...
+### Aliases
+
+```
+pipelinerun, pr, pipelineruns
+```
+
+### Available Commands
+
+```
+describe    Describe a pipelinerun in a namespace
```
Comment (joaedwar) [inline docs/cmd/tkn_pipelinerun.md]: Is it PipelineRun, pipelinerun, or Pipelinerun

It should be consistent when referring to it.

File: docs/cmd/tkn_pipelinerun.md [hunk 23]
```diff
@@ -0,0 +1,41 @@
...
+
+### Options inherited from parent commands
+
+```
+--azure-container-registry-config string   Path to the file containing Azure container registry configuration information.
+```
+
+### SEE ALSO
+
+* [tkn pipelinerun list](tkn_pipelinerun_list.md)	 - List all pipelineruns in given namespace.
```
Comment (joaedwar) [inline docs/cmd/tkn_pipelinerun.md]:
```suggestion
[tkn pipelinerun list](tkn_pipelinerun_list.md)	 - Lists all `PipelineRuns` in a given namespace.
```

Comment (joaedwar) [inline docs/cmd/tkn_pipelinerun.md]: Outside of the CLI command it looks like PipelineRun, Task, etc... are presented as `PipelineRun`, `Task`, etc...

File: docs/cmd/tkn_pipelinerun_describe.md [hunk 24]
```diff
@@ -0,0 +1,38 @@
...
+### Aliases
+
+```
+describe, desc
+```
+
+### Options
+
+```
+--allow-missing-template-keys   If true, ignore any errors in templates when a field or map key is missing in the template. Only applies to golang and jsonpath output formats. (default true)
```
Comment (joaedwar) [inline docs/cmd/tkn_pipelinerun_describe.md]: Edit like earlier example (Line 22)

File: docs/cmd/tkn_task_list.md [hunk 25]
```diff
@@ -0,0 +1,38 @@
...
+### Aliases
+
+```
+list, ls
+```
+
+### Options
+
+```
+--allow-missing-template-keys   If true, ignore any errors in templates when a field or map key is missing in the template. Only applies to golang and jsonpath output formats. (default true)
```
Comment (joaedwar) [inline docs/cmd/tkn_task_list.md]: See previous comments on this statement

File: docs/cmd/tkn_taskrun_list.md (hunk 15 above)
Comment (joaedwar) [inline docs/cmd/tkn_taskrun_list.md]:
```suggestion
Lists taskruns in a namespace.
```

File: docs/cmd/tkn_taskrun_list.md [hunk 26]
```diff
@@ -0,0 +1,38 @@
...
+### Aliases
+
+```
+list, ls
+```
+
+### Options
+
+```
+--allow-missing-template-keys   If true, ignore any errors in templates when a field or map key is missing in the template. Only applies to golang and jsonpath output formats. (default true)
```
Comment (joaedwar) [inline docs/cmd/tkn_taskrun_list.md]: Refer to earlier comments on this statement

File: README.md [hunk 27]
```diff
@@ -5,6 +5,38 @@
...
+
+## Need to add installation steps.
+
+### Utilities
+
+These are commands that provide some useful information to the user.
+
+ * The `tkn help` command displays a list of the commands with helpful information.
+ * The [`tkn completion` command](docs/cmd/tkn_completion.md) will output a BASH completion script for `tkn` to allow command completions with tabs.
+ * The [`tkn version` command](docs/cmd/tkn_version.md) will output the version of cli.
```
[the remark this answers is not in the document]
Comment (pradeepitm12) [reply README.md]: Changed the pattern like below 
 * [`tkn completion:`](docs/cmd/tkn_completion.md) Outputs a BASH completion script for `tkn` to allow command completion with Tab.

File: docs/cmd/tkn_pipelinerun.md (hunk 22 above)
Comment (pradeepitm12) [reply docs/cmd/tkn_pipelinerun.md]: the command is  pipelinerun

File: docs/cmd/tkn_pipelinerun.md (hunk 23 above)
Comment (pradeepitm12) [reply docs/cmd/tkn_pipelinerun.md]: We can go with `pipelinerun`, `task`, etc.. to maintain consistency.

**Pull request merged**